crates/optimism/consensus/src/lib.rs

```diff
@@ -98,13 +101,42 @@ impl<ChainSpec: EthChainSpec + OpHardforks, B: Block> Consensus<B>
             return Err(ConsensusError::BodyTransactionRootDiff(error.into()))
         }
 
-        // EIP-4895: Beacon chain push withdrawals as operations
+        // Check empty shanghai-withdrawals
         if self.chain_spec.is_shanghai_active_at_timestamp(block.timestamp()) {
-            validate_shanghai_withdrawals(block)?;
+            canyon::verify_empty_shanghai_withdrawals(block.body()).map_err(|err| {
+                trace!(target: "op::consensus",
+                    block_number=block.number(),
+                    %err,
+                    "block failed validation",
+                );
+
+                ConsensusError::Other
+            })?;
+        } else {
+            return Ok(())
         }
```

Comment:
it's a bit weird that we put this into canyon if this is a shanghai specific check

Reply:
can make a new file, np